models/montecarlo.py

```diff
@@ -0,0 +1,94 @@
+from models.model import Model
+from datetime import datetime
+from environment import Environment, preprocess_data
+import numpy as np
+import pandas as pd
+import random
+import matplotlib.pyplot as plt
+
+'''
+A monte carlo model with epsilon greedy policy.
+
+- Episode: A complete run of the model from the start to the end of the training data.
+- Epsilon: The probability of choosing a random action instead of the best action for exploring instead of exploiting.
+- State-Action Dictionary: A dictionary that stores the average reward and count of each state-action pair.
+
+'''
+
+class MonteCarlo(Model):
+    def __init__(self, cleaned_data, actions, epsilon=0.1):
+        super().__init__(state_space=None, action_space=actions)
+        self.cleaned_data = cleaned_data
+        self.actions = actions
+        self.epsilon = epsilon
+        self.state_action_dict = self.initialize_policy()
+
+    def initialize_policy(self):
+        return {(state, action): (0, 0) 
+                for state in ['11', '10', '01', '00'] 
+                for action in range(len(self.actions))}
+
+    def choose_action(self, state):
+        if random.uniform(0, 1) < self.epsilon:
+            return random.randint(0, len(self.actions) - 1)  # Explore
+        return np.argmax(self.calculate_average_reward(state)) # Exploit
+
+    def update_policy(self, state, action, reward):
+        reward_sum, count = self.state_action_dict[(state, action)]
+        self.state_action_dict[(state, action)] = (reward_sum + reward, count + 1)
+
+    def train(self, episodes, train_env):
+        n_steps = len(train_env.data)
+
```

Comment:
To have a better generalization as stated in the paper, please use random starting points to create each episode. You can see my implementation as a reference.